File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Messages/MessageContact.cs
using System;
using System.Collections.Generic;

namespace VecompSoftware.DocSuiteWeb.Entity.Messages
{

    public class MessageContact : DSWBaseEntity
    {
        #region [ Constructor ]

        public MessageContact() : this(Guid.NewGuid()) { }

        public MessageContact(Guid uniqueId)
            : base(uniqueId)
        {
            MessageContactEmail = new HashSet<MessageContactEmail>();
        }
        #endregion

        #region [ Properties ]

        /// <summary>
        /// Tipo di contatto
        /// </summary>
        public MessageContactType ContactType { get; set; }
        /// <summary>
        /// Contatto di posizione
        /// <example>Sender, Recipient, RecipientCc, RecipientBcc</example>
        /// </summary>
        public MessageContactPosition ContactPosition { get; set; }
        /// <summary>
        /// Descrizione della mail
        /// </summary>
        public string Description { get; set; }

        #endregion

        #region [ Navigation Properties ]
        /// <summary>
        /// Messaggio in cui è contenuto l'attachment
        /// </summary>
        public virtual Message Message { get; set; }

        public virtual ICollection<MessageContactEmail> MessageContactEmail { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Message/MessageContact.vb
Imports VecompSoftware.DocSuiteWeb.Data

<Serializable()>
Public Class MessageContact
    Inherits AuditableDomainObject(Of Int32)
    Enum ContactTypeEnum
        User = 1
        Role = 2
        AdGroup = 3
    End Enum

    Enum ContactPositionEnum
        Sender = 1
        Recipient = 2
        RecipientCc = 3
        RecipientBcc = 4
    End Enum

#Region " Properties "

    Public Overridable Property Message As DSWMessage

    Public Overridable Property ContactType As ContactTypeEnum

    Public Overridable Property ContactPosition As ContactPositionEnum

    Public Overridable Property Description As String

    Public Overridable Property ContactEmails As IList(Of MessageContactEmail)

#End Region

#Region " Constructors "

    Public Sub New()

    End Sub

    Public Sub New(contactType As ContactTypeEnum, description As String, contactPosition As ContactPositionEnum)
        Me.ContactType = contactType
        Me.Description = description
        Me.ContactPosition = contactPosition
    End Sub

#End Region

End Class
